rust/main/hyperlane-base/src/contract_sync/cursors/mod.rs

```diff
@@ -60,6 +62,7 @@ impl Indexable for MerkleTreeInsertion {
             HyperlaneDomainProtocol::Fuel => todo!(),
             HyperlaneDomainProtocol::Sealevel => CursorType::SequenceAware,
             HyperlaneDomainProtocol::Cosmos => CursorType::SequenceAware,
+            HyperlaneDomainProtocol::Ton => CursorType::SequenceAware, // need check
```

Comment:
Merkle tree insertion should be `SequenceAware` as it is. So, comment `need check` can be removed.